### DISCUSSION TITLE
iOS Dev Here - Interest in a dedicated NVR app for iPhone/iPad?

### DISCUSSION THREAD

**Discussion by evanheckert** [blakeblackshear/frigate#4002] [Ideas]
I've used both Home Assistant and Blue Iris forever, and have always longed for a friendlier, more aesthetic UI for the Blue Iris mobile app. I've played with Unifi Protect, but I'm not interested in the lock-in to their cameras, and using a proxy to get my PoE cams into it has been buggy at best. I've been following Frigate since it first came on the scene, and I'm thrilled that Blake and other contributors have continued to double down on it.

I appreciate the Home Assistant integrations, and do use them. However, my personal preference is to have a dedicated NVR app on my phone. I've built several CRUD apps both for web and iOS, and I'm wanting to scratch the itch of a great NVR iPhone app, even if just for myself. 

That said, if there's interest to others, I'd love for my efforts to bring value to the community that's given me so much value.

**If there is interest, what would you like to see?** I'd imagine the biggest gains could be around:

- native UI gestures and navigation, not limited to web-compatible API's
- more control of notifications, being focus-mode aware, etc.
- deep linking direct to the camera feed
- geofencing without needing Home Assistant
- Home/Lock Screen widgets
- editing config from UI (now in 0.12)

EDIT 3/28/2023: It seems to me that there are three main problems to solve with a good mobile experience:
1. **Off-network access**: How to follow best practices while making it as simple as possible for the less-techy person to configure. What are examples that we can model after, ie. Nabu Casa's hosting, or Blue Iris's router setup walkthrough and QR code configuration?
2. **User Authentication**: @hagak mentioned OpenID Connect, which I'll need to look more into. Curious what will be needed on the frigate/server side, as the goal will be for everything to work natively with Frigate, and not needed additional services on the server side in order to use the iOS app.
3. **Performant Video Streaming & Timeline Loading/Scrubbing**: This is the core of the experience, and will likely be a learning curve for me. I'm curious to what extent I'll need to understand compression and streaming techniques vs just plugging into an RTSP stream.

**Comment by NickM-27** [comment]
There will likely be interest, someone made an android app recently which had good interest as well. 

As far as the editing in the UI, there's an open feature request issue for it. I only started contributing in the last 9 months, but as far as I understand it just hasn't been a priority. Also just to add currently the docs, addon install, etc all have the config file as read only so if you add UI configuration it'll need to have users update their docker config to make the config file read/write (I don't think addon could be supported until we change that in an update)

I think one day frigate will offer UI configuration as well.

**Reply by evanheckert** [reply]
That makes sense. I'll look into the existing issue and chime in there on that front, as I'm interesting in contributing there as well.

Have you seen any discussion or interest around an 'official' iOS/mobile app? I'm open to doing my own thing like the Android app you referenced, but I'd much rather build it in a way that builds towards where Frigate is going, and not just as a wrapper that reacts to where it is at the moment.

**Reply by NickM-27** [reply]
To be honest I'm not sure what the difference would be between a community app and official app. Given that frigate is open source it's not like an official app would have any more access to the API. 

Always happy to have contributions so if you ever are wondering the direction of frigate or if something you're working on aligns with the frigate roadmap I'd be happy to discuss (and I believe Blake would be as well). 

For context I'm an android dev professionally, but trying to get away from my day job and work on other things, I'm sure I could offer some opinions on the app but not much on the code itself. I certainly don't speak for Blake or the project, but my own point of view is that there's nothing that would make the app more official than the frigate Lovelace card is "official". As it is open source, there is always room for other options (other iOS apps).

I could totally be missing context so let me know if you disagree.

**Reply by evanheckert** [reply]
I definitely see your point, and really appreciate that aspect of open source. 

I'm likely missing more context than you are - I see mention of a documented roadmap [here](https://github.com/blakeblackshear/frigate/discussions/2338), but not sure if that turned up anywhere I could read?

There's ~~two~~ ~~three~~ 3.5 angles where I see an 'official' app as valuable:

1. **Trust**. People tend to have much more confidence and trust in mobile apps that are built by (or at least endorsed by) the organization that makes the thing. My experiences have shown this is especially true with native mobile apps, as they expose the user more than web - from device APIs, tracking, notifications spam, features getting paywalled, etc.
2. **Timing**.  Unofficial apps can create quite a headache when either trying to get ahead of the main project, or if they fall behind. Inevitably it creates more issues/support requests on both sides, and things break. If it was an official app, I would imagine compatibility & feature parity (where appropriate) would be an important checkbox before major releases.
3. **Money**.  Ugh, money. Not even open source can escape the need to consider it. This is another item where understanding the long term vision might eliminate any considerations here. If Blake or Frigate as an organization are able to make money from the project (and I hope they do!), then it's important to the org what people are paying for, and what value they receive. 
(3.5. **Control**.) If this is my project, people are paying me to use it, and they want a native app, I don't want them to have to go pay more money for some random person's mobile app, just to have an experience that I can't control. What if it's buggy? What if it removes features they're paying for, or worse, adds its own paywall to them? Not good. 
So what to do? I'd want to develop it in-house, and make it available either for free, or as an additional revenue stream that helps to fund the project directly. But most importantly, I can control/direct the paying customer's experience front-to-back. If something goes wrong, I can fix it, and I can make sure I'm delivering the value that I want to deliver.

That's a fairly raw brain-dump of why I thought an 'official' app is worth exploring. Again, there may be an established roadmap that renders those points moot - or perhaps a lack of interest/demand will do so. Till then, if people are open to talking it out, I'm certainly here for it!

**Reply by NickM-27** [reply]
Good points and I see where you're coming from. The other side of the coin though is that it's unclear how it would be supported by current frigate contributors. To be clear, I mean whoever made an iOS app which was deemed "official". In the scenario that the person decided to or was unable to continue working on it for whatever reason, now that app has no one working on it and we don't have time (and potentially don't have the ability) to work on both frigate proper and the iOS app. It could also be a bad user experience if this happened.

In any case, my personal opinion is it makes sense to start as a community app and at some point in the future once things are fleshed out and community response is understood it would make sense to revisit the "official" label and what that would mean. (again just my 2¢)

I'm not sure the current roadmap is documented somewhere.

**Reply by evanheckert** [reply]
“ In any case, my personal opinion is it makes sense to start as a community app and at some point in the future once things are fleshed out and community response is understood it would make sense to revisit the "official" label and what that would mean. (again just my 2¢)”

I agree that this makes the most sense.

**Comment by LordNex** [comment]
> There will likely be interest, someone made an android app recently which had good interest as well. 
> 
> As far as the editing in the UI, there's an open feature request issue for it. I only started contributing in the last 9 months, but as far as I understand it just hasn't been a priority. Also just to add currently the docs, addon install, etc all have the config file as read only so if you add UI configuration it'll need to have users update their docker config to make the config file read/write (I don't think addon could be supported until we change that in an update)
> 
> I think one day frigate will offer UI configuration as well. 

I love the idea. This is on great peice of software that just keeps getting better. If you start an iOS development, I'll drop a few bucks in the coffers when I can

**Comment by LordNex** [comment]
> I've used both Home Assistant and Blue Iris forever, and have always longed for a friendlier, more aesthetic UI for the Blue Iris mobile app. I've played with Unifi Protect, but I'm not interested in the lock-in to their cameras, and using a proxy to get my PoE cams into it has been buggy at best. I've been following Frigate since it first came on the scene, and I'm thrilled that Blake and other contributors have continued to double down on it.
> 
> 
> 
> I appreciate the Home Assistant integrations, and do use them. However, my personal preference is to have a dedicated NVR app on my phone. I've built several CRUD apps both for web and iOS, and I'm wanting to scratch the itch of a great NVR iPhone app, even if just for myself. 
> 
> 
> 
> That said, if there's interest to others, I'd love for my efforts to bring value to the community that's given me so much value.
> 
> 
> 
> **If there is interest, what would you like to see?** I'd imagine the biggest gains could be around:
> 
> 
> 
> - native UI gestures and navigation, not limited to web-compatible API's
> 
> - more control of notifications, being focus-mode aware, etc.
> 
> - deep linking direct to the camera feed
> 
> - geofencing without needing Home Assistant
> 
> - Home/Lock Screen widgets
> 
> - editing config from UI ?***
> 
> 
> 
> *** I need to look into more of why this isn't in the web UI. I'm betting Blake has thoroughly thought that through and there's good reason to keep it as a YAML file. I certainly am not looking to work against the philosophies that have been making this such a well loved project.  

Might check out DoubleTake for home assistant. It ties Frigate to Facial Recognition, works through MQTT, and runs in Home Assistant

**Reply by evanheckert** [reply]
I'm curious how integral this is to the Frigate experience, and whether it's a feature set that will either be integrated or officially supported in some capacity? I think it's an amazing extension of Frigate's functionality, but supporting an app that needs to follow one project has enough complexity that I'm not sure two would be responsible to attempt.

Though perhaps the app itself wouldn't need to?  I'll have to explore the different levels of integration that might make sense.

**Edit**: As I think about this, this is an example where the extendable model of Home Assistant really shines. It allows separate but related projects to interop without the two needing to strictly coordinate or account for each other. This point might be the single biggest case **against** a dedicated iOS app, as it either has to chase multiple projects, or handicap its functionality to be less useful than using the Home Assistant add on(s). I'm open to ideas/counterarguments here.

**Comment by jefftidwell-lab** [comment]
I would be willing to donate $2500 to get an application that is stable and has same functionality as Unifi Protect app.

**Reply by evanheckert** [reply]
What are the key features of the Unifi app that they do particularly well that you'd like to see here? I have a Cloud Key Gen2 Plus running Protect, but unifi-cam-proxy is currently broken and I don't have a Unifi camera at the moment to fully experience/reverse engineer it properly.

**Comment by blakeblackshear** [comment]
Just want to chime in and say that I am supportive of this idea. There isn't currently a mobile app because I just don't have time for it. I don't know much about the landscape in mobile development, but it would be nice to see something cross platform. I generally look to home assistant for patterns that seem to work, and they ultimately brought in apps more officially. I imagine that would make sense here too.

**Reply by evanheckert** [reply]
@blakeblackshear I appreciate the feedback (and Frigate as a whole, of course). 

Cross-platform is doable - I use React Native for mobile, which is designed to maximize code reuse across platforms while doing as much natively as possible. The production apps I've built have only been deployed on iOS, but Android is on the roadmap for one of them anyway. I'll focus on iOS initially here, but at least the tech stack won't back us into a corner.

Where would be an appropriate place to explore/dialogue about the roadmap of Frigate? I'm outlining an MVP, and would love to skate to where the proverbial puck is going as much as possible.

**Reply by NickM-27** [reply]
https://github.com/blakeblackshear/frigate/pull/4055 outlines ideas for the next release

**Comment by LanceWesterhoff** [comment]
We're in the midst of retiring an old Q-see NVR and we're considering Frigate as a possible replacement. The lack of a mobile app is a major impediment. It would be one thing if it were just me but there are "less technical" people in the household that also use the Q-see app daily for remote viewing (yes, it still works - even though Q-see is no more). We also have the other problem that different [even less technical] people need access to only a subset of available cameras on the property. Q-see allows different login accounts with different permissions and does a generally good job with this requirement.

So in short, yes this sounds like a great idea. And I would be happy to beta test & contribute if we end up switching to Frigate (vs other NVR platforms we're considering). As for other features, in addition to the need for login accounts/permissions (which is likely more fundamental to Frigate vs. an app alone), since we use the app exclusively as we don't even login to the NVR, we also like to be able to review past clips/events in the Q-see app via a timeline function.

**Reply by evanheckert** [reply]
Thanks for outlining your needs. I think you're correct that accounts and permissions logic would need to exist in the core Frigate codebase. Then the mobile app could only access the cameras that the logged in user's account had permissions to see.

**Reply by ajatoledo** [reply]
A Frigate iOS app would be great, and I'd be interested in supporting it in whatever way possible. 

Currently, for a basic interface to Frigate, I use an [iOS website shortcut](https://support.apple.com/guide/iphone/bookmark-favorite-webpages-iph42ab2f3a7/ios), but by no means is it a replacement for an app.

**Reply by evanheckert** [reply]
@ajatoledo Thanks for chiming in. Curious what you’re doing when you’re off network?

**Reply by ajatoledo** [reply]
@evanheckert Off network, as in away from home? I use tailscale with my router as a tailscale subnet router to access the Frigate feed. It's not perfect, but it allows me to access the camera feeds.

**Reply by hagak** [reply]
Just to provide more insight on providing auth. into Frigate you can put frigate behind a reverse proxy like Traefik and have a rule for some form of auth.  Such as user/password or better OIDC (which is what I do).  So if a mobile app could support say OIDC that would be perfect.

**Comment by yehpo** [comment]
Just jumping in to say that I’d love to see an iOS app. I’m not a developer so probably can’t help very much towards the project unfortunately, but I would definitely buy the app once it’s launched. :-)

**Comment by DrSpaldo** [comment]
I would like to see the dedicated app as well... As @ajatoledo does, I also use the iOS website shortcut and use external access via Cloudflare authentication.

**Comment by saltyautomation** [comment]
Have you thought of writing it in Flutter and Dart to get iOS, Android, Web, and Desktop app all in one single codebase https://flutter.dev/ ? There is a wrapper for video playback plugin with controls here https://pub.dev/packages/chewie . 

**Edit**
There is a recently released sample app that I was just playing around with that could provide some inspiration on many ways to navigate snapshots and videos. https://github.com/gskinnerTeam/flutter-wonderous-app you can also download the wonderous app from the play store or apple store.

**Reply by evanheckert** [reply]
I'm a react-native dev, which touts similar cross-platform capabilities. The focus initially will be iOS, but if well received, will likely leverage those cross platform capabilities long-term.

**Reply by norbertjoni** [reply]
I would like to try and test the dev app when it's will be possible.

**Reply by dschrade94** [reply]
How far into development is the app?

**Reply by evanheckert** [reply]
@dschrade94 I have the following working locally:

* Live streaming with WebRTC
* View events
* In events, toggle between snapshot and video clip
* Read only view of config in collapsible tree JSON hierarchy 

Currently working on making a video playback zoomable and full screen capable, both for live view and clips. 

First goal is feature parity with the Web UI, then locally saving the server connection, then will likely open to beta testing.

**Reply by ccutrer** [reply]
Sounds fantastic. I'd be happy to be a beta tester. My current setup is that my frigate is available directly via HTTPS on the internet, with [vouch proxy](https://github.com/vouch/vouch-proxy) handling authentication for me. I have a ~30 cams, mostly 4MP and 4k, a mix of AVC and HEVC, and a UniFi Doorbell Cam.

My priorities for a mobile app, and things I miss from the Blue Iris app, are super-fast loading of the stream (which Frigate Web UI currently has with WebRTC -- once you've found the correct cam in the UI), quickly drilling down to a particular cam, and easy navigation between cams (I miss swiping left and right between cams in a group from BI). One thing I really dislike about the Frigate Web UI (in general, but on mobile in particular) is how much space is wasted in margins and padding. On the main camera list I can only see 2.5 cameras at a time (https://www.dropbox.com/s/d35hartoecbyko9/Screenshot%202023-04-17%20at%209.10.02%20AM.png?dl=0), and the sidebar that pulls out when viewing a particular camera, there's so much space between labels I can only see half of my list (https://www.dropbox.com/s/7dhbqoq9szt6ly9/Screenshot%202023-04-17%20at%209.08.49%20AM.png?dl=0). Oh, and zoom! I can't zoom in to a (often paused) video in the WebUI to try and identify a small object. It looks like you're working on that already, thank you 🙇!

**Reply by dschrade94** [reply]
I'd also gladly beta test.

**Reply by Chluz** [reply]
Same here, very interested in this app

**Reply by ideal2545** [reply]
Happy to beta test too and help provide some funding

**Reply by norbertjoni** [reply]
Happy to test it

**Reply by kyle4269** [reply]
> @dschrade94 I have the following working locally:
> 
> 
> 
> * Live streaming with WebRTC
> 
> * View events
> 
> * In events, toggle between snapshot and video clip
> 
> * Read only view of config in collapsible tree JSON hierarchy 
> 
> 
> 
> Currently working on making a video playback zoomable and full screen capable, both for live view and clips. 
> 
> 
> 
> First goal is feature parity with the Web UI, then locally saving the server connection, then will likely open to beta testing. 

I'll test also if you need more.

**Reply by doubleopinter** [reply]
Is this still a thing? I’m happy to test something as well, been looking for a nice simple iOS app as well.

**Reply by mitsu98** [reply]
currently working with this using Swift

![ts](https://github.com/blakeblackshear/frigate/assets/87440954/1f6f85c7-3e35-4ea0-ac77-327713819ef3)

**Reply by Jacou** [reply]
Looks very slick.

**Reply by rwpi** [reply]
Looks great! If you need any testers, I'd like to help.

**Reply by ideal2545** [reply]
> currently working with this using Swift
> 
> 
> 
> ![ts](https://github.com/blakeblackshear/frigate/assets/87440954/1f6f85c7-3e35-4ea0-ac77-327713819ef3)
> 
> 
> 
> 
> 
> 
> 
> 
> 
> 

Looks awesome!!

**Reply by Go2Engle** [reply]
Do you have any eta for this? It looks so good!

**Comment by yehpo** [comment]
That looks amazing, can’t wait to test it when it is ready!

**Comment by dschrade94** [comment]
Looks amazing

V/R

Dalton M Schrader
________________________________

**Comment by citgot** [comment]
Looking forward to see if there is some development for a dedicated Frigate app. I'd like to get the similar experience of the Blue Iris app but with Frigate instead. Using HA on iPhone isn't nearly as quick as a dedicated app for cameras. I'd be willing to pay for the app. My use case would not need all the Frigate settings in the app. Just viewing live cameras, recordings, alerts and some settings regarding motion/object detection, notifications and video settings would be enough to begin with.

**Comment by toadee** [comment]
Awesome to see something coming from this.

Recently stumbled upon the Android app which my partner has and was then tried to see if there was an ios version as I'm apple. Home Assistant is great but my biggest issue is the inability to go full screen.

**Comment by billyjacoby** [comment]
an iOS app is the last thing I need to make the switch to solely using Frigate as my NVR solution.

@evanheckert If you'd like any help with this I'd love to help the development efforts. I'm a full time React Native developer so I think I could be a big help here. Looking forward to hearing back!

**Reply by evanheckert** [reply]
I'd be interested in help, or even handing it off, as I got busy preparing for twins due in a couple months! I'd love to be able to flesh it out myself, but no sense in me being the bottleneck if others are capable and willing to carry the torch!

**Reply by billyjacoby** [reply]
I'm happy to take a look at it and see what kind of bandwidth I can dedicate now, feel free to send me a link to the repo either via invite or DM me on [Discord](https://discordapp.com/users/billyjacoby/7369) or [Twitter](https://twitter.com/billyjacoby)

**Reply by domneedham** [reply]
Hi both, another dev here. Work with Xamarin forms but have used React on web. Want an iOS app enough I'm willing to give it a try (even just the odd contribution where possible).

Username on discord is domneedham

**Reply by evanheckert** [reply]
Well isn't this delightful. I did a clean wipe of my drive a couple weeks ago, not realizing I hadn't pushed to github. So I have a half finished app on my phone, and no code. Ugh, the amount of time I spent getting webRTC to work... I'm sorry gents, I inadvertently torched our head start. 😒

**Reply by billyjacoby** [reply]
One of the many reasons I commit to git wayyyy too often 😂

I'm on holiday for the next 2ish weeks, but when I get back I can hopefully try and get started here. If someone beats me to it, be sure to post a repo here - otherwise I'll post mine when I get to it!

**Comment by jaco** [comment]
Is it really worth developing custom app for iOS only? Hybrid apps on today’s hardware are really indistinguishable from native ones (if done right of course). 

Frigate front end is already spa application. Imo it will be much better to direct the effort to improve mobile experience and build on top of that (for example using capacitor)…

**Reply by billyjacoby** [reply]
Yeah so if I'm starting from scratch it'll be both iOS and Android via React Native.

**Reply by evanheckert** [reply]
Agreed. React native makes this fairly straightforward.

**Comment by DrSpaldo** [comment]
I was wondering if anyone had suggestions for an app to use (just to view the streams) in the meantime before a real one is made? I can view the stream on the PC fine but on the mobile/iOS it often isn’t a great experience on Edge or Safari on the iPhone

**Reply by kirsch33** [reply]
frigate-hass-card but if you don’t already use Home Assistant it would be quite a bit of setup

**Comment by billyjacoby** [comment]
Just wanted to give an update here, I finally had a few minutes to get the initial skeleton of the app scaffolded out in React Native this weekend. There's not much else to report yet, but just wanted to keep everyone posted that I am beginning to actively (nights and weekends only for now) work on this.

Will post back here with updates when there's something significant to share. Cheers!

**Reply by billyjacoby** [reply]
If anyone has experience with streaming WebRTC video to React Native, please feel free to reach out! 

I'm struggling to get the initial proof of concept even hammered out.

**Comment by billyjacoby** [comment]
@evanheckert Running into some weird WebRTC/Websocket stuff here, I know you mentioned some pain points with this so wondering if you could spare a few minutes to potentially shed some light on what those issues may have been?

For some more info here it looks like all of the WS messages that I'm sending and receiving from my Frigate server match what I'm seeing in the browser console, but I still can't get a video stream up and playing.

If anyone else has experience with WebRTC and React Native please feel free to lend a hand! The file where I've tried implementing this can be found in the repo [here](https://github.com/billyjacoby/bird-watcher/blob/main/src/components/webRTCpoc.tsx)!

**Reply by evanheckert** [reply]
I spent 90% of my time on this issue, as using the same methods as the web version didn't work. I spent a few hours going back and forth with ChatGPT, which eventually helped find something that worked. I just searched through my chat history, and I guess it must have been before they saved chat history, sadly. Come to think of it, I think I used bing chat, which I don't think saves chat history at all.

**Comment by domneedham** [comment]
If you share the repo I’ll try and take a look myself. Not got any
experience with full RN dev but do mobile dev daily and have used react
before so should get up and running fine.

**Reply by billyjacoby** [reply]
That would be awesome. Repo can be found here: https://github.com/billyjacoby/bird-watcher/blob/main/src/components/webRTCpoc.tsx

I linked to the specific component where i’m trying to get WebRTC working.

**Reply by domneedham** [reply]
Hey, taken a look today. You got another way of messaging you about it?

**Reply by billyjacoby** [reply]
> Hey, taken a look today. You got another way of messaging you about it?

Discord is probably easiest: billyjacoby

**Comment by billyjacoby** [comment]
Okay I've made good progress here today. I'm able to get streams showing up and working properly - but *only sometimes*..?

The thing is that it's super intermittent. Sometimes the stream loads and plays fine, but other times it won't. I found myself reloading in the bundler a whole bunch of times and only getting a proper stream maybe 20% of the time? 

I'm thinking that it has to be some sort of race condition somewhere, or me not properly handling the `RCTPeerConnection` websocket messages.

I'd love if someone could take a look at this and see if there's something glaringly wrong here. As long as you add your local frigate server to the `.env` file in the repo (here)[https://github.com/billyjacoby/bird-watcher] you should be able to scroll through the WebRTC feeds after building the app locally.

Hoping for some help here sooner rather than later!

**Comment by domneedham** [comment]
Got it cloned and running this morning (painful setup but got there in the
end). Seeing the same, intermittent loading. Will try and take a look when
I can.

**Comment by danfairhurst** [comment]
Would love to get involved in this. Will try and get it compiled and running locally. Never worked on iOS apps before so might be a bit of a learning curve

**Reply by billyjacoby** [reply]
Awesome! I'm happy to help if you get stuck anywhere. If you've got experience with web dev and specifically React it should look pretty familiar. The initial React Native setup is for sure the hardest part 😅

**Comment by billyjacoby** [comment]
This is fully underway. Going to open a new discussion for more visibility once this is ready for a bit more testing.

I could definitely use help with the WebRTC and JSMPEG streaming aspects if anyone's got the experience!

**Reply by DrSpaldo** [reply]
Very keen for this, I have been considering switching to Blue Iris for the NVR aspect and just using Frigate for the detection because of the lack of app :(

**Comment by dschrade94** [comment]
Please post link here

V/R

Dalton M Schrader
________________________________

**Reply by billyjacoby** [reply]
Sorry I updated this comment after you replied. I'm gonna hold off on posting until I have a fully testable version. Hoping to get there in a week or so. If you have dev experience you can check it out here though:

https://github.com/billyjacoby/bird-watcher

**Reply by kiloptero** [reply]
amazing job!!

**Reply by TheOriginal92** [reply]
Hey @billyjacoby any Update on the App? Im willing to dive into react/iOS Development if it helps! Is there a discord or something?

**Comment by dschrade94** [comment]
No development experience. Just beta testing and also a systems integrator

V/R

Dalton M Schrader
________________________________

**Comment by Zandiexoder** [comment]
I would be very excited to see the app have similar functionality to the nest app, mainly a decent footage scrub which is missing on the home assistant.

**Comment by Dyad-ITSS** [comment]
Not sure if you've take a look at some like Netbird.io or other WireGuard mesh network solutions but it might help to have that as the basis for the connection from the app to Frigate server baked in so for non-tech people they wouldn't have to worry about port forwarding, reverse proxies or DDNS. 

"It just works" approach.

https://www.wireguard.com/embedding/

https://openziti.io/

**Reply by Zandiexoder** [reply]
Yeah it definitely would not work remotely without some setup, but all I want is an app to scrub through footage and look at live view of cameras. Also Tailscale is a great easy VPN to use with home assistant to access it remotely.

**Reply by Dyad-ITSS** [reply]
100%, I was more putting it forward as an idea to have frigate and iOS / Android app have an overlay network built-in so for the everyday user who doesn't really know enough about networking, VPN's, port forward etc won't have to worry about that part when configuring for remote use. Just connect the app via a QR code displayed on Frigate UI and the app auto-connects when not on the same LAN / IP range as the Frigate server.

**Reply by billyjacoby** [reply]
Yeah that would be entirely separate infrastructure that is without a doubt outside the scope of this project 😅

**Reply by Zandiexoder** [reply]
Yeah definitely adds a layer of completion not needed at the time, though I was more thinking along the lines of just putting it in the docs on how to set it up, and not imbed it in the software.

**Comment by NoOneReally2021** [comment]
Sign me up for beta 😃

**Comment by OnFreund** [comment]
I'd definitely love to see that.
One thing that I think is important to many is support of HTTP basic auth. Since Frigate doesn't have auth built-in, many people use a reverse proxy with basic auth, and on mobile browsers you're required to type in your password too often.
An app could store the credentials.

**Reply by ccutrer** [reply]
Or not-Basic auth. I use Vouch Proxy, which just looks like a redirect to a different domain.

**Comment by alekslyse** [comment]
Did anything happen with this project? I would love to be on the testflight for testing and maybe contributing code for a project like this

**Reply by mjehrhart** [reply]
Take a look at [ViewU](https://www.viewu.app), a new open-source community-based iOS app designed for Frigate NVR. The app entered beta testing on TestFlight on May 17th, 2024. ViewU aims to enhance the user experience for those utilizing Frigate by offering features such as a timeline of events, live RTSP feeds, and push notifications with images.

If you're interested in testing the app and providing feedback, you can join the beta testing program via TestFlight. For more information, visit the [ViewU official website](https://www.viewu.app) or check out the relevant discussions and updates on Reddit and other community forums.

**Reply by LexiconCode** [reply]
Is there anything feature equivalent for Android?

**Reply by NickM-27** [reply]
Frigate now supports being installed as a PWA which means it functions just like any other app. Frigate 0.15 adds support for native notifications as well.

https://docs.frigate.video/configuration/pwa

**Reply by TheOriginal92** [reply]
@NickM-27 how do I enable native notifications?

**Reply by NickM-27** [reply]
https://deploy-preview-13787--frigate-docs.netlify.app/configuration/notifications

**Reply by TheOriginal92** [reply]
@NickM-27 thank you! I’m running version 0.15.0-e76f4e9 and can not find a settings option for notifications.

**Reply by NickM-27** [reply]
Are you connecting with a secure connection?

**Reply by TheOriginal92** [reply]
@NickM-27 seems like it’s just http. How do I use https in a local environment?

**Reply by NoOneReally2021** [reply]
Following

**Comment by billyjacoby** [comment]
All of my progress can be found in the GH repo here: https://github.com/billyjacoby/bird-watcher

I haven't had much time lately to get into it, but the next thing I want to do is turn the camera views into regular webviews into the web interface to make that a little more robust.

**Comment by MikePaer** [comment]
Thanks for your effort so far! Is anyone still working on this? An iOS-optimized app would be amazing to have. My biggest gripe with Frigate is the poor experience in iOS browsers. It seems the options are Frigate in mobile browser (not great) or Frigate Lovelace card in HomeAssistant (a little better but still not great on iOS due to lack of full screen button for live views etc.)

**Reply by billyjacoby** [reply]
I'm plugging away at this slowly, but I'm largely stumped on the native WebRTC implementation. The best solution I have right now is to render the browser in app in order to view live streams.

Past events seem to be working pretty well though. I just haven't had the time to clean anything up in a while.

**Reply by NickM-27** [reply]
just wanted to mention that for the next version of frigate the UI is currently undergoing a rewrite from the ground up with a new design and new features, there is definitely a focus on making the mobile experience great as well.

> (a little better but still not great on iOS due to lack of full screen button for live views etc.)

this is a safari limitation, no one can fix this but Apple

**Reply by DrSpaldo** [reply]
> just wanted to mention that for the next version of frigate the UI is currently undergoing a rewrite from the ground up with a new design and new features, there is definitely a focus on making the mobile experience great as well.
> 
> > (a little better but still not great on iOS due to lack of full screen button for live views etc.)
> 
> this is a safari limitation, no one can fix this but Apple

@NickM-27 will there be a way to zoom in on the live view to a specific place in the camera? rather than just the middle. That is a massive one for me 🙏

**Reply by NickM-27** [reply]
yes, zooming has been implemented

**Reply by fpytloun** [reply]
Can’t be live streams implemented same way as Homeassistant Companion app does with frigate camera card? It is currently best option for iOS.  Seems native, supporting picture in picture and zooming too.
 
![image](https://github.com/blakeblackshear/frigate/assets/948799/bea004ab-9e5a-41eb-949e-503bdc74418e)

**Reply by NickM-27** [reply]
The home assistant companion app is just a web view

**Reply by MikePaer** [reply]
The best live camera streaming experience I’ve seen on iOS is an app called “IPCams” by Nogosoft LLC. That’s what I’d suggest to draw inspiration from for the Frigate native iOS app live experience.

**Reply by billyjacoby** [reply]
> The home assistant companion app is just a web view

Do you know or have a link to what the URL of the WebView is? That seems like it would be a real easy add then

**Reply by OnFreund** [reply]
> Do you know or have a link to what the URL of the WebView is? That seems like it would be a real easy add then
For the HA app? It's just the regular url you would use in your browser

**Reply by f1d094** [reply]
> > The home assistant companion app is just a web view
> 
> Do you know or have a link to what the URL of the WebView is? That seems like it would be a real easy add then

The source URL from frigate depends on what type of stream you set the card to pull from (e.g. webrtc or jsmpeg) and if you are leveraging go2rtc restreaming vs direct rtsp sources. Looking at the source for the HASS Frigate Lovelace Card is probably your best bet...

If you have not setup HASS + Frigate Lovelace Card, I would urge you to do so. The Frigate Lovelace Card is by far the most versatile interface to a given Frigate camera. Zoom-able, fullscreen-able, supports two-way audio, and so on.

**Reply by billyjacoby** [reply]
I briefly looked at the source code last week but haven't had a chance to dive in deeper unfortunately.

I think it's more complicated than just a source URL, but I could be wrong. If that's it than this is an easy add that makes my app almost ready for testflight.

I'm totally open to contributions if that's something you think you could get in!

**Reply by NickM-27** [reply]
yes, for mse and webrtc it is not as simple as just dropping a source url in. There is connection negotiation via websocket and then either that websocket is used (mse) or a peer connection is created (pc)

**Reply by billyjacoby** [reply]
> yes, for mse and webrtc it is not as simple as just dropping a source url in. There is connection negotiation via websocket and then either that websocket is used (mse) or a peer connection is created (pc)

@NickM-27 I don't know why but I've had nothing but trouble getting the WebRTC negotiation working in RN for this app. I basically copy and pasted the code from the web React app into the React Native app [here](https://github.com/billyjacoby/bird-watcher/blob/main/src/components/WebRTCView/WebRTCView.tsx#L68) but this doesn't seem to work reliably - and doesn't work at all over a VPN.

I'd super appreciate if you could take a quick look at the implementation in that file and let me know if anything jumps out to you as immediately wrong in some way.

I think the next best option here is to just add the lovelace card into a webview, but I know that'll take me a bit of time to get to.

**Comment by AlexKLMan** [comment]
I would love an app!
Idea, critical alerts if someone goes into a zone during a time; so it will wake me up.

**Reply by mjehrhart** [reply]
A new open-source app named ViewU is being developed for Frigate NVR, and it has just entered beta testing on TestFlight as of May 17th, 2024. This app, while not an official iOS app for Frigate, is designed to enhance the user experience by providing an intuitive and functional interface for iOS devices. Key features of ViewU include an event timeline, live RTSP feeds, and push notifications with images. 

The beta testing phase already has a number of users, but the developers are looking for more participants to provide feedback and help improve the app. 

If you're interested in joining the beta testing program, you can visit the [Viewu official website](https://www.viewu.app) for more information, or check out the related [Reddit thread](https://www.reddit.com/r/frigate_nvr/comments/1cu9xeu/ios_app_viewu_is_now_in_open_beta_for_testflight/) for updates and details on how to get involved.  

https://github.com/mjehrhart/viewu
 
https://installation.viewu.app
